smdebug/tensorflow/base_hook.py

```diff
@@ -298,17 +299,25 @@ def _get_writers(self, tensor_name, tensor_ref) -> List[FileWriter]:
                 # if device str is empty or cpu in worker
                 if not bool(worker) or "CPU" in worker:
                     if self.save_all_workers:
-                        return list(self.writer_map.values())
+                        if shape_writers is False:
+                            return list(self.writer_map.values())
+                        else:
+                            return list(self.shape_writer_map.values())
                     else:
-                        return [self.writer_map[self.device_map[self.chief_worker]]]
+                        if shape_writers is False:
+                            return [self.writer_map[self.device_map[self.chief_worker]]]
+                        elif self._saving_shapes_in_step():
+                            return [self.shape_writer_map[self.device_map[self.chief_worker]]]
                 elif self.save_all_workers or worker == self.chief_worker:
-                    return [self.writer_map[self.device_map[worker]]]
-            elif self.writer:
+                    if shape_writers is False:
+                        return [self.writer_map[self.device_map[worker]]]
+                    elif self._saving_shapes_in_step():
```

Comment:
Why do we need a different shape writer?
If self._saving_shapes_in_step() is True for every step, do we end up using only shape_writer?
Does it not make sense then to have a single writer and move the is_saving_shape logic to the Writer class?

Reply:
Simplified it by merging it with writer